Source/Core/Core/IOS/Network/KD/Mail/WC24Send.cpp

```diff
@@ -61,6 +80,137 @@ bool WC24SendList::CheckSendList() const
   return true;
 }
 
+u32 WC24SendList::GetNumberOfMail() const
+{
+  return Common::swap32(m_data.header.number_of_mail);
+}
+
+u32 WC24SendList::GetEntryId(u32 entry_index) const
+{
+  return Common::swap32(m_data.entries[entry_index].id);
+}
+
+u32 WC24SendList::GetMailSize(u32 index) const
+{
+  return Common::swap32(m_data.entries[index].msg_size);
+}
+
+ErrorCode WC24SendList::DeleteMessage(u32 index)
+{
+  ErrorCode error = NWC24::DeleteFileFromVFF(NWC24::Mail::SEND_BOX_PATH, GetMailPath(index), m_fs);
+  if (error != WC24_OK)
+    return error;
+
+  // Fix up the header then set the entry to 0.
+  m_data.header.number_of_mail = Common::swap32(Common::swap32(m_data.header.number_of_mail) - 1);
+  m_data.header.next_entry_offset = Common::swap32(128 + (index * 128));
+  m_data.header.next_entry_id = Common::swap32(Common::swap32(m_data.header.next_entry_id) - 1);
```

Comment:
This feels wrong to me; if you have three messages with ids 0, 1, 2 and the header having at next_entry_id = 3 and delete the message with id 1, you can't just decrement the next_entry_id to 2 -- it's still in use.

Reply:
Or wait, id 0 stands for 'no mail here', right? But same difference.